Frontend/src/pages/registrations/components/RegistrationList.jsx

```diff
@@ -1,101 +1,98 @@
-import React, { useState } from 'react'
-import deleteRegistration from '../../../api/registration/delete/delete_registration'
-import getRegistrations from '../../../api/registration/get/get_registrations'
-import updateRegistration from '../../../api/registration/patch/update_registration'
+import { NonInteractiveTable } from '@thewca/wca-components'
+import React, { useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import getCompetitionInfo from '../../../api/competition/get/get_competition_info'
+import { getConfirmedRegistrations } from '../../../api/registration/get/get_registrations'
+import getCompetitorInfo from '../../../api/user/get/get_user_info'
 import styles from './list.module.scss'
-import StatusDropdown from './StatusDropdown'
-
-function RegistrationRow({
-  competitorId,
-  eventIDs,
-  serverStatus,
-  competitionID,
-  setRegistrationList,
-  registrationList,
-}) {
-  const [status, setStatus] = useState(serverStatus)
-
-  return (
-    <tr>
-      <td>{competitorId}</td>
-      <td>{eventIDs.join(',')}</td>
-      <td>
-        <StatusDropdown status={status} setStatus={setStatus} />
-      </td>
-      <td>
-        <button
-          onClick={(_) => {
-            updateRegistration(competitorId, competitionID, status)
-          }}
-        >
-          {' '}
-          Apply
-        </button>
-      </td>
-      <td>
-        <button
-          onClick={(_) => {
-            deleteRegistration(competitorId, competitionID)
-            setRegistrationList(
-              registrationList.filter((r) => r.competitor_id !== competitorId)
-            )
-          }}
-        >
-          Delete
-        </button>
-      </td>
-    </tr>
-  )
-}
 
 export default function RegistrationList() {
```

Comment:
I think some/most of the state in here, like header and footer, can just be computed from other state instead of having its own state. In fact, there could probably be just two states, for `competitionInfo` and `registrations` with everything else derived from these. This seems safer, because theoretically all these current states could get out-of-sync as they are, whereas if they're calculated from the single source(s) of truth then they're guaranteed to be in-sync. And separate constants could be calculated separately, instead of all crammed into a couple of large use effects (which could just receive the data and call setters).
Generally, only using state for something that needs to be stateful is a good rule of thumb, I think.

Reply:
Done!